runtime/src/drivers.rs

```diff
@@ -166,15 +169,17 @@ impl Drivers {
     }
 
     fn validate_dpe_structure(mut drivers: &mut Drivers) -> CaliptraResult<()> {
-        let dpe = &drivers.persistent_data.get().dpe;
-        let root_idx = Self::get_dpe_root_context_idx(dpe)?;
-        // Ensure context/TCI tree is well-formed (all nodes reachable from root and no cycles)
-        if !dpe.validate_context_tree(root_idx) {
+        let dpe = &mut drivers.persistent_data.get_mut().dpe;
+        let dpe_validator = DpeValidator { dpe };
```

Comment:
Even then we would still have to create a DpeValidator, pass the DPE to it, and call the validation functions so I feel like it would be leaking a bit of the logic in the DpeValidator to DpeInstance.